### PR TITLE
Enable Offboard Bodyrate control for Rovers

### PR DESCRIPTION
**Pull request by Jaeyoung-Lim**
**Describe problem solved by this pull request**
Previously, rover steering control has been mainly done in a open-loop fashion by simply setting the steering angle from the controller.

However, this results in unstable and unpredictable steering behaviors in high speed as well as uneven terrain. This open loop control also have been shown to introduce oscillations when using on boats. 

**Describe your solution**
This PR adds a bodyrate controller (yaw) that can be activated through offboard control on the rover. This is mainly to demonstrate the rate control behaviors on the rover.

This also moves the [ECL_Controller](https://github.com/PX4/Firmware/blob/master/src/modules/fw_att_control/ecl_controller.h) baseclass that was recently moved to the firmware from PX4/ecl to a library, so that it can be used in the `rover_pos_controller` as well as the `fw_att_controller`

**Test data / coverage**
Offboard control log: https://review.px4.io/plot_app?log=a30ebfd4-96cb-4fc8-9e55-8606fb5ef4b5

**Additional Context**
Switching to a body rate based controller will bring the following benefits
- Introduce stabilized mode to rover
- Deprecate the parameter `GND_MAX_ANG` and minimize the difference between the differential rover and ackerman steering vehicle (https://github.com/PX4/Firmware/issues/13468)

### CODE REVIEW

File: src/modules/rover_pos_control/RoverAttitudeControl.cpp
Comment (julianoes) [inline src/modules/rover_pos_control/RoverAttitudeControl.cpp:35]: Lies! I would just delete these `@file` as they tell you nothing and are more often wrong than right.

File: src/modules/rover_pos_control/RoverAttitudeControl.cpp
Comment (julianoes) [inline src/modules/rover_pos_control/RoverAttitudeControl.cpp:53]: This needs to be throttled or removed, otherwise the spam might lead to more problems.

File: src/modules/rover_pos_control/RoverAttitudeControl.cpp
Comment (julianoes) [inline src/modules/rover_pos_control/RoverAttitudeControl.cpp:3]:
```suggestion
 *   Copyright (c) 2013-2016 Estimation and Control Library (ECL). All rights reserved.
 *   Copyright (c) 2017-2020 PX4 Development Team. All rights reserved. 
```

And is this source actually from ECL and it's now copied into the PX4 source? I'm a bit confused.

File: src/modules/rover_pos_control/RoverAttitudeControl.hpp
Comment (julianoes) [inline src/modules/rover_pos_control/RoverAttitudeControl.hpp:35]: Same.

File: src/modules/rover_pos_control/RoverPositionControl.cpp
Comment (julianoes) [inline src/modules/rover_pos_control/RoverPositionControl.cpp:370]: When? I assume this will be done before a merge?

File: src/modules/rover_pos_control/RoverPositionControl.cpp
Comment (julianoes) [inline src/modules/rover_pos_control/RoverPositionControl.cpp:392]:
```suggestion
	control_input.lock_integrator = (current_speed < _param_rate_i_minspeed.get());
```

File: src/modules/rover_pos_control/RoverAttitudeControl.cpp
Comment (Jaeyoung-Lim) [reply src/modules/rover_pos_control/RoverAttitudeControl.cpp:3]: Yes, it was moved over from https://github.com/PX4/Firmware/pull/14284

File: src/modules/rover_pos_control/RoverPositionControl.cpp
Comment (Jaeyoung-Lim) [reply src/modules/rover_pos_control/RoverPositionControl.cpp:392]: Oh, this is much better!

File: src/modules/rover_pos_control/RoverAttitudeControl.cpp
Comment (Jaeyoung-Lim) [reply src/modules/rover_pos_control/RoverAttitudeControl.cpp:54]: Is there already an API where I can throttle messages? This is actually coming from https://github.com/PX4/Firmware/blob/master/src/modules/fw_att_control/ecl_pitch_controller.cpp#L56

File: src/modules/rover_pos_control/RoverPositionControl.cpp
Comment (Jaeyoung-Lim) [reply src/modules/rover_pos_control/RoverPositionControl.cpp:370]: I was a bit hesitant on doing this before running it on a real vehicle, since somebody would be using the offboard attitude control setpoints

Comment (julianoes) [reply src/modules/rover_pos_control/RoverPositionControl.cpp:370]: Ok, your call.

**Review by julianoes — DISMISSED** [review]
Not opposed :smile: thanks @Jaeyoung-Lim

File: src/modules/rover_pos_control/RoverPositionControl.cpp
Comment (dagar) [inline src/modules/rover_pos_control/RoverPositionControl.cpp:414]: We should switch these to uORB::Subscription at some point.

Comment (Jaeyoung-Lim) [reply src/modules/rover_pos_control/RoverPositionControl.cpp:414]: Right, I will do it on a separate PR

**Comment by dagar** [comment]
I actually don't think it's worth preserving the "ECL" controllers. I think if we manually inlined the subset of methods it could all be collapsed into a relatively tiny amount of code.

For example the multicopter rate controller. https://github.com/PX4/Firmware/blob/8763d71bf0b603e0da42ea16fceae7fd2590546f/src/modules/mc_rate_control/RateControl/RateControl.cpp#L63-L67

There are also some in progress FW modifications here. https://github.com/PX4/Firmware/pull/15256

**Comment by Jaeyoung-Lim** [comment]
@dagar In that case, should I move the base class into the Roverattitude controller?

**Comment by dagar** [comment]
> @dagar In that case, should I move the base class into the Roverattitude controller?

That works for now unless you want to absorb it entirely. A common PID class or even rate controller could make sense at some point, I just think we can do better than that ECL code.

**Comment by LorenzMeier** [comment]
@Jaeyoung-Lim What's the state here?

**Comment by Jaeyoung-Lim** [comment]
@LorenzMeier Sorry for dragging this on. I will wrap this up this week

**Pull request closed without merging**
